poly/src/polynomial/multivariate/sparse.rs

```diff
@@ -0,0 +1,408 @@
+//! A sparse multivariate polynomial represented in coefficient form.
+use crate::{
+    multivariate::{SparseTerm, Term},
+    MVPolynomial, Polynomial,
+};
+use ark_ff::Field;
+use ark_std::{
+    cmp::Ordering,
+    fmt,
+    ops::{Add, AddAssign, Neg, Sub, SubAssign},
+    vec::Vec,
+};
+use rand::Rng;
+
+#[cfg(feature = "parallel")]
+use rayon::prelude::*;
+
+/// Stores a sparse multivariate polynomial in coefficient form.
+#[derive(Derivative)]
+#[derivative(Clone, PartialEq, Eq, Hash, Default)]
+pub struct SparsePolynomial<F: Field, T: Term> {
+    /// The number of variables the polynomial supports
+    #[derivative(PartialEq = "ignore")]
+    pub num_vars: usize,
+    /// List of each term along with its coefficient
+    pub terms: Vec<(F, T)>,
+}
+
+impl<F: Field, T: Term> SparsePolynomial<F, T> {
+    fn remove_zeros(&mut self) {
+        self.terms.retain(|(c, _)| !c.is_zero());
+    }
+}
+
+impl<F: Field> Polynomial<F> for SparsePolynomial<F, SparseTerm> {
+    type Point = Vec<F>;
+
+    /// Returns the zero polynomial.
+    fn zero() -> Self {
+        Self {
+            num_vars: 0,
+            terms: Vec::new(),
+        }
+    }
+
+    /// Checks if the given polynomial is zero.
+    fn is_zero(&self) -> bool {
+        self.terms.is_empty() || self.terms.iter().all(|(c, _)| c.is_zero())
+    }
+
+    /// Returns the total degree of the polynomial
+    fn degree(&self) -> usize {
+        self.terms
+            .iter()
+            .map(|(_, term)| (*term).degree())
```

Comment:
I think this should work?
```suggestion
            .map(|(_, term)| term.degree())
```